lib/astc_encode.cpp

```diff
@@ -0,0 +1,963 @@
+        case KTX_PACK_ASTC_BLOCK_6x6x5: return VK_FORMAT_ASTC_6x6x5_UNORM_BLOCK_EXT;
+        case KTX_PACK_ASTC_BLOCK_6x6x6: return VK_FORMAT_ASTC_6x6x6_UNORM_BLOCK_EXT;
+        }
+    }
+
+    return VK_FORMAT_ASTC_6x6_SRGB_BLOCK; // Default is 6x6 sRGB image
+}
+
+/**
+ * @memberof ktxTexture
+ * @ingroup write
+ * @~English
+ * @brief Creates valid astc encoder action from string.
+ *
+ * @return      Valid astc_profile from string
+ */
+static astcenc_profile
+astcEncoderAction(const ktxAstcParams &params, const uint32_t* bdb) {
+
+    if (params.function == KTX_PACK_ASTC_ENCODER_FUNCTION_SRGB &&
+        params.mode == KTX_PACK_ASTC_ENCODER_MODE_LDR) {
+        return ASTCENC_PRF_LDR_SRGB;
+    }
+    else if (params.function == KTX_PACK_ASTC_ENCODER_FUNCTION_LINEAR &&
+        params.mode == KTX_PACK_ASTC_ENCODER_MODE_LDR) {
+        return ASTCENC_PRF_LDR;
+    }
+    else if (params.function == KTX_PACK_ASTC_ENCODER_FUNCTION_LINEAR &&
+        params.mode == KTX_PACK_ASTC_ENCODER_MODE_HDR) {
+        return ASTCENC_PRF_HDR;
+    }
+    else if (params.function == KTX_PACK_ASTC_ENCODER_FUNCTION_UNKNOWN) {
+        // If no options provided assume the user wants to use
+        // color space info provided from the file
+
+        ktx_uint32_t transfer = KHR_DFDVAL(bdb, TRANSFER);
+        if (transfer == KHR_DF_TRANSFER_SRGB &&
+            params.mode == KTX_PACK_ASTC_ENCODER_MODE_LDR)
+            return ASTCENC_PRF_LDR_SRGB;
+        else if (transfer == KHR_DF_TRANSFER_LINEAR) {
+            if (params.mode == KTX_PACK_ASTC_ENCODER_MODE_LDR)
+                return ASTCENC_PRF_LDR;
+            else
+                return ASTCENC_PRF_HDR;
+        }
+    }
+    // TODO: Add support for the following
+    // KTX_PACK_ASTC_ENCODER_ACTION_COMP_HDR_RGB_LDR_ALPHA; not supported
+
+  return ASTCENC_PRF_LDR_SRGB;
+}
+
+
+/**
+ * @memberof ktxTexture
+ * @ingroup write
+ * @~English
+ * @brief Creates valid astc encoder swizzle from string.
+ *
+ * @return      Valid astcenc_swizzle from string
+ */
+static astcenc_swizzle
+astcSwizzle(const ktxAstcParams &params) {
+
+    astcenc_swizzle swizzle{ASTCENC_SWZ_R, ASTCENC_SWZ_G, ASTCENC_SWZ_B, ASTCENC_SWZ_A};
+
+    std::vector<astcenc_swz*> swizzle_array{&swizzle.r, &swizzle.g, &swizzle.b, &swizzle.a};
+
+    for (int i = 0; i < 4; i++) {
+        if (params.inputSwizzle[i] == 'r')
+            *swizzle_array[i] = ASTCENC_SWZ_R;
+        else if (params.inputSwizzle[i] == 'g')
+            *swizzle_array[i] = ASTCENC_SWZ_G;
+        else if (params.inputSwizzle[i] == 'b')
+            *swizzle_array[i] = ASTCENC_SWZ_B;
+        else if (params.inputSwizzle[i] == 'a')
+            *swizzle_array[i] = ASTCENC_SWZ_A;
+        else if (params.inputSwizzle[i] == '0')
+            *swizzle_array[i] = ASTCENC_SWZ_0;
+        else if (params.inputSwizzle[i] == '1')
+            *swizzle_array[i] = ASTCENC_SWZ_1;
+    }
+
+    return swizzle;
+}
+
+static void
+astcBlockSizes(ktx_uint32_t block_size,
```

Comment:
As I said in an earlier comment, don't use size, use dimension for clarity and consistency with the data format descriptor spec an d Vulkan spec. terminology.

Reply:
Fixed in https://github.com/KhronosGroup/KTX-Software/pull/433/commits/c01d05d4cfa0353f848c18051d90ba94862115b5